TestingHomework-Discounts/Repositories/PromoRepository.cs

```diff
@@ -2,18 +2,26 @@
 using System;
 using System.Collections.Generic;
 using System.Text;
+using TestingHomework.Tests.Data;
+using TestingHomework.Tests.DTOs;
+using TestingHomework_Discounts.Models;
 
 namespace TestingHomework_Discounts
 {
-    class PromoRepository : DbContext
+    public class PromoRepository : DbContext
     {
-        public DbSet<PromoCode> PromoCodes { get; set; }
-        public DbSet<Product> Products { get; set; }
-        public DbSet<User> Users { get; set; }
+        public DbSet<PromoCodeDTO> PromoCodes { get; set; }
```

Comment:
What is the reasoning for having DTOs and the normal domain models?

DTOs are meant to decouple the domain models from the data source. DTOs are exclusively used to interact with the database and then they are mapped to Domain models. DTOs then are then never used above data source access.